src/main/java/loqor/ait/core/engine/SubSystem.java

```diff
@@ -0,0 +1,190 @@
+package loqor.ait.core.engine;
+
+import java.lang.reflect.Type;
+import java.util.function.Supplier;
+
+import com.google.gson.*;
+
+import loqor.ait.api.Disposable;
+import loqor.ait.api.Initializable;
+import loqor.ait.api.TardisComponent;
+import loqor.ait.client.tardis.ClientTardis;
+import loqor.ait.core.engine.impl.*;
+import loqor.ait.core.tardis.ServerTardis;
+import loqor.ait.core.tardis.Tardis;
+import loqor.ait.core.tardis.manager.ServerTardisManager;
+import loqor.ait.data.Exclude;
+import loqor.ait.data.enummap.Ordered;
+
+public abstract class SubSystem extends Initializable<TardisComponent.InitContext> implements Disposable {
```

Comment:
Imo it should get its own init context.

Reply:
itd be identical tho

Reply:
Well, currently, maybe, yes.
But it may change in future and the class' signature will change.
If addons used the old (current) and we update it to have its own context the plugins wont work.

Reply:
true that